engine/src/main/java/io/camunda/zeebe/engine/state/instance/DbJobState.java

```diff
@@ -543,4 +545,23 @@ private boolean isValidForRestore(final JobRecordValue jobRecord) {
     final var job = jobRecord.getRecord();
     return job.getRecurringTime() > -1 && job.getRetries() > 0;
   }
+
+  /**
+   * Removing jobs that are mistakenly inserted into the backoff column family due to <a
+   * href="https://github.com/camunda/zeebe/issues/15954">this issue</a>
+   */
+  private void removeWrongJobs() {
+    final var failedKeys = getFailedJobKeys();
+    final var backoffJobs = getBackoffJobKey();
+
+    backoffJobs.retainAll(failedKeys);
```

Comment:
:wrench: I'm not sure that we need to do this since `backoffJobs` is always a subset of `failedJobs`, i.e. any job added in the `backoffColumnFamily` must be in state `FAILED`.

@nicpuppa are you doing this in case a job that is not `FAILED` ended up in the `backoffColumnFamily`?

Reply:
> @nicpuppa are you doing this in case a job that is not `FAILED` ended up in the `backoffColumnFamily`?

It should not happen, so I will remove the retain